posthog/temporal/batch_exports/squash_person_overrides.py

```diff
@@ -75,32 +75,42 @@
 """
 
 CREATE_JOIN_TABLE_FOR_DELETES_QUERY = """
-CREATE TABLE {database}.person_overrides_to_delete ON CLUSTER {cluster}
+CREATE OR REPLACE TABLE {database}.person_overrides_to_delete ON CLUSTER {cluster}
 ENGINE = Join(ANY, LEFT, team_id, distinct_id) AS
 SELECT
-    team_id, distinct_id, groupUniqArray(_partition_id) AS partitions
+    team_id,
+    distinct_id,
+    sum(person_id != dictGet('{database}.{dictionary_name}', 'person_id', (team_id, distinct_id))) AS total_unsquashed,
+    sum(person_id = dictGet('{database}.{dictionary_name}', 'person_id', (team_id, distinct_id))) AS total_squashed
```

Comment:
In the event of concurrent dictionary reload, the new `person_id` won't match the one used in `SQUASH_EVENTS_QUERY`, so it will count as a `total_unsquashed`, making it non-zero and ultimately not deleted (which appears to be correct).

Maybe renaming the aliases to `pending_person_overrides` and `applied_person_overrides` would work better? Do you have any aliases in mind?

Reply:
Going to merge this to run some tests, but happy to revisit the aliases later!